manifests/plugin/dynflow.pp

```diff
@@ -0,0 +1,35 @@
+# = Foreman Proxy Dynflow plugin
+#
+# This class installs Dynflow support for Foreman proxy
+#
+# === Parameters:
+#
+# $enabled::         Enables/disables the plugin
+#                    type:boolean
+#
+# $listen_on::       Proxy feature listens on https, http, or both
+#
+# $group::           Owner of plugin configuration
+#
+class foreman_proxy::plugin::dynflow (
+  $enabled           = $::foreman_proxy::plugin::dynflow::params::enabled,
+  $listen_on         = $::foreman_proxy::plugin::dynflow::params::listen_on,
+  $group             = $::foreman_proxy::plugin::dynflow::params::group,
+) inherits foreman_proxy::plugin::dynflow::params {
+
+  if (versioncmp($::rubyversion, '1.9.3') < 0) {
+    fail('Ruby 1.8.7 is not supported by the Dynflow plugin.')
+  }
```

Comment:
With other cases like these we've chosen not to error out in puppet, for example where there were no Debian packages available during release for the reason you question it. Right now I'd leave this out.